src/app/tests/TestDataModelSerialization.cpp

```diff
@@ -16,70 +16,53 @@
  *    limitations under the License.
  */
 
-/**
- *    @file
- *      This file implements unit tests for CHIP Interaction Model Command Interaction
- *
- */
-
 #include <app-common/zap-generated/cluster-objects.h>
 #include <app/data-model/Decode.h>
 #include <app/data-model/Encode.h>
 #include <lib/core/TLV.h>
 #include <lib/support/CHIPMem.h>
-#include <lib/support/UnitTestRegistration.h>
-#include <nlunit-test.h>
 #include <system/SystemPacketBuffer.h>
 #include <system/TLVPacketBufferBackingStore.h>
 
+#include <lib/core/StringBuilderAdapters.h>
+#include <pw_unit_test/framework.h>
+
 namespace {
 
 using namespace chip;
 using namespace chip::app;
 using namespace chip::app::Clusters;
 
-class TestDataModelSerialization
+class TestDataModelSerialization : public ::testing::Test
 {
 public:
-    static void TestDataModelSerialization_EncAndDecSimpleStruct(nlTestSuite * apSuite, void * apContext);
-    static void TestDataModelSerialization_EncAndDecSimpleStructNegativeEnum(nlTestSuite * apSuite, void * apContext);
-    static void TestDataModelSerialization_EncAndDecNestedStruct(nlTestSuite * apSuite, void * apContext);
-    static void TestDataModelSerialization_EncAndDecNestedStructList(nlTestSuite * apSuite, void * apContext);
-    static void TestDataModelSerialization_EncAndDecDecodableNestedStructList(nlTestSuite * apSuite, void * apContext);
-    static void TestDataModelSerialization_EncAndDecDecodableDoubleNestedStructList(nlTestSuite * apSuite, void * apContext);
-
-    static void TestDataModelSerialization_OptionalFields(nlTestSuite * apSuite, void * apContext);
-    static void TestDataModelSerialization_ExtraField(nlTestSuite * apSuite, void * apContext);
-    static void TestDataModelSerialization_InvalidSimpleFieldTypes(nlTestSuite * apSuite, void * apContext);
-    static void TestDataModelSerialization_InvalidListType(nlTestSuite * apSuite, void * apContext);
-
-    static void NullablesOptionalsStruct(nlTestSuite * apSuite, void * apContext);
-    static void NullablesOptionalsCommand(nlTestSuite * apSuite, void * apContext);
-
-    void Shutdown();
+    static void SetUpTestSuite() { ASSERT_EQ(chip::Platform::MemoryInit(), CHIP_NO_ERROR); }
+    static void TearDownTestSuite()
+    {
+        Shutdown();
```

Comment:
Maybe just calls `mStore.Release();` here without any "Shutdown" wrapper. Also, I wonder why `mStore` is static, is it shared between tests (if that's the case, that's somehow not ideal, because unit tests might depend on each other)?

Reply:
- well in order to be able to call `mStore.Release();` in TearDownTestSuite it has to be static.
- also the reason I called Shutdown instead of directly putting `mStore.Release();` is:
     1. `mStore.Release();` has a return of `CHECK_RETURN_VALUE` so  it needs to be assigned to the `System::PacketBufferHandle  buf` object
     2. the `buf` object will be destructed when we reach the end of scope of `TearDownTestSuite()` just after `chip::Platform::MemoryShutdown();`  --> this results in an error because the destruction of `buf` will attempt to free the buffer, but since `memoryInitialized` of `CHIPMem-Malloc.cpp` is Zero due to `chip::Platform::MemoryShutdown();` , we hit a VerifyorDie and exception happens....maybe I can explicitly Free the buffer before memoryshutdown...

Reply:
ok I added `buf = nullptr;` just before MemoryShutdown to avoid the error that happens

Reply:
> well in order to be able to call mStore.Release(); in TearDownTestSuite it has to be static.

That's exactly the point. Is there any particular reason why it's called in the `TearDownTestSuite()` and not in the `TearDown()`. Like I've said, it smells like the buffer is shared between unit test cases, which might be not good.

Reply:
done.